openmc/mgxs/mgxs.py

```diff
@@ -919,23 +919,6 @@ def load_from_statepoint(self, statepoint):
                   'linked with a summary file'
             raise ValueError(msg)
 
-        # Override the domain object that loaded from an OpenMC summary file
-        # NOTE: This is necessary for micro cross-sections which require
-        # the isotopic number densities as computed by OpenMC
```

Comment:
So I did some searching and figured out why there is such a huge performance hit. In the past, this method queried the `Summary` object referenced by the `StatePoint` for each `Cell`, `Material`, etc. In the past, the `Summary` contained a collection of all instances of each class, and could quickly return an instance with a particular ID without first recursing through the entire CSG tree each time.

The `Summary` no longer contains these collections of instances, and instead only includes a reference to a `Geometry`. I propose adding back in collections of `Materials`, `Cells`, `Universes`, `Lattices` so we can revert to much more quickly querying the `Summary` for instances by ID. What do you think @paulromano?